src/common/src/types/to_text.rs

```diff
@@ -14,33 +14,64 @@
 
 use std::fmt::Write;
 
-use super::{DatumRef, ScalarRefImpl};
+use chrono::{TimeZone, Utc};
+
+use super::{DataType, DatumRef, ScalarRefImpl};
 
 // Used to convert ScalarRef to text format
 pub trait ToText {
+    // to_text impl to_text_with_type using default type of ScalarRefImpl.
     fn to_text(&self) -> String;
+    fn to_text_with_type(&self, ty: DataType) -> String;
 }
 
-macro_rules! implement_using_to_string {
-    ($({ $scalar_type:ty } ),*) => {
+macro_rules! implement_using_to_sql {
+    ($({ $scalar_type:ty , $data_type:ident} ),*) => {
         $(
             impl ToText for $scalar_type {
                 fn to_text(&self) -> String {
                     self.to_string()
                 }
+                fn to_text_with_type(&self, ty: DataType) -> String {
+                    match ty {
+                        DataType::$data_type => self.to_text(),
+                        _ => unreachable!(),
+                    }
+                }
             }
         )*
     };
 }
 
-implement_using_to_string! {
-    { i16 },
-    { i32 },
-    { i64 },
-    { String },
-    { &str },
-    { crate::types::OrderedF32 },
-    { crate::types::OrderedF64 }
+implement_using_to_sql! {
+    { i16,Int16  },
+    { i32,Int32  },
+    { &str,Varchar },
+    { crate::types::OrderedF32,Float32 },
+    { crate::types::OrderedF64,Float64 }
+}
```

Comment:
Note the conflict with #6611. We now use `itoa` for integers and `ryu` for floats.